src/azul/terra.py

```diff
@@ -216,25 +233,23 @@ def check_api_access(self, source: TDRSource) -> None:
             if total == 0:
                 raise self._insufficient_access(resource)
             elif total == 1:
-                snapshot_id = one(response['items'])['id']
-                endpoint = self._repository_endpoint(tdr_path, snapshot_id)
-                response = self.oauthed_http.request('GET', endpoint)
-                if response.status == 200:
-                    response = json.loads(response.data)
-                    # FIXME: Response now contains a reference to the Google
-                    #        project name in a property called `dataProject`.
-                    #        Use this approach (or reuse this code) to avoid
-                    #        hardcoding the project ID.
-                    #        https://github.com/DataBiosphere/azul/issues/2504
-                    log.info('TDR client is authorized for API access to %s: %r', resource, response)
-                else:
-                    assert False, snapshot_id
+                return one(response['items'])['id']
             else:
-                raise RuntimeError('Ambiguous response from TDR API', endpoint)
+                raise RequirementError('Ambiguous response from TDR API', endpoint)
         elif response.status == 401:
             raise self._insufficient_access(endpoint)
         else:
-            raise RuntimeError('Unexpected response from TDR API', response.status)
+            raise RequirementError('Unexpected response from TDR API', response.status)
+
+    def _get_source_response(self, source: TDRSource) -> bytes:
```

Comment:
This doesn't look right. There is some duplication between this and snapshot_id_from_source. I think this wants to be refactored into a more canonical form.